legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java

```diff
@@ -169,84 +198,123 @@ protected void serializeOutput(MutableMap<FileGenerationSpecification, List<Gene
+    }
+
+    private <T extends PackageableElement> void filterPackageableElementsByIncludes(Map<String, T> elementsByPath) throws MojoExecutionException
     {
         if (this.inclusions != null)
         {
-            ResolvedGenerationSpecificationFilter filter;
+            ResolvedPackageableElementFilter filter;
             try
             {
-                filter = resolveGenerationSpecFilter(this.inclusions);
+                filter = resolvePackageableElementFilter(this.inclusions, elementsByPath.keySet());
             }
             catch (Exception e)
             {
-                throw new MojoExecutionException("Error resolving included GenerationSpecifications", e);
-            }
-            generationSpecsByPath.keySet().removeIf(filter.negate());
-            if ((filter.generationSpecPaths != null) && Iterate.anySatisfy(filter.generationSpecPaths, p -> !generationSpecsByPath.containsKey(p)))
-            {
-                throw new MojoExecutionException(LazyIterate.reject(filter.generationSpecPaths, generationSpecsByPath::containsKey).makeString("Could not find included GenerationSpecifications: ", ", ", ""));
+                throw new MojoExecutionException("Error resolving included PackageableElements", e);
             }
+            elementsByPath.keySet().removeIf(filter.negate());
         }
     }
 
-    private void filterGenerationSpecsByExcludes(Map<String, GenerationSpecification> generationSpecsByPath) throws MojoExecutionException
+    private <T extends PackageableElement> void filterPackageableElementsByExcludes(Map<String, T> elementsPath) throws MojoExecutionException
     {
-        if ((this.exclusions != null) && !generationSpecsByPath.isEmpty())
+        if ((this.exclusions != null) && !elementsPath.isEmpty())
         {
-            ResolvedGenerationSpecificationFilter filter;
+            ResolvedPackageableElementFilter filter;
             try
             {
-                filter = resolveGenerationSpecFilter(this.exclusions);
+                filter = resolvePackageableElementFilter(this.exclusions, elementsPath.keySet());
             }
             catch (Exception e)
             {
-                throw new MojoExecutionException("Error resolving excluded GenerationSpecifications", e);
+                throw new MojoExecutionException("Error resolving excluded PackageableElements", e);
             }
-            generationSpecsByPath.keySet().removeIf(filter);
+            elementsPath.keySet().removeIf(filter);
         }
     }
 
-    private static ResolvedGenerationSpecificationFilter resolveGenerationSpecFilter(GenerationSpecificationFilter generationSpec) throws Exception
+    private static ResolvedPackageableElementFilter resolvePackageableElementFilter(PackageableElementFilter elementFilter, Set<String> elementPaths) throws Exception
     {
-        Set<String> generationSpecPaths;
-        if (generationSpec.directories != null)
+        Set<String> elementsPaths;
```

Comment:
It's confusing to have a parameter named "elementPaths" and a local variable named "elementsPaths". For that matter, "elementsPaths" is a confusing name on its own.